### ISSUE TITLE
mio regressed using a debugger with ipc-channel

### ISSUE THREAD

**Issue by jdm** [servo/ipc-channel#131]
I ended up doing a build of Servo using ipc-channel master, and discovered that hitting a breakpoint in a debugger caused a ton of "System call interrupted" panics from ipc-channel after resuming. We're going to need to address this before we can use any revision past 357abb967bc4c703af5d4a9f95e2958317e9e8a1 in Servo.

**Comment by jdm** [comment]
cc @danlrobertson

**Comment by danlrobertson** [comment]
You mention `mio` in the issue, but `mio` was not used in the commit mentioned. Did you use `git bisect` to determine that 357abb9 was at fault? If it was the introduction of `mio` that is causing this, the at fault commit would be f6cefc5650e5cca57629f3a9610c20dcc0289fa7.

**Comment by jdm** [comment]
f6cefc5650e5cca57629f3a9610c20dcc0289fa7 makes the panics appear, and be882855182a40e1cbfe60418cd8f87f3278346b makes them disappear.

**Comment by jdm** [comment]
(those are the two most recent commits according to `git log --topo-order`)

**Comment by jdm** [comment]
My original comment could have been written more clearly; I did not mean to indicate that 357abb9 was the commit at fault.

**Comment by danlrobertson** [comment]
Thanks that's super helpful! I'll look into it ASAP.

**Comment by jdm** [comment]
Did you figure anything out, @danlrobertson?

**Comment by danlrobertson** [comment]
I did some work on this last night, and after some digging around carllerche/mio#494 and https://github.com/servo/ipc-channel/blob/master/src/platform/unix/mod.rs#L475-L478 look like likely culprits. Tonight I'm going to try ignoring `EINTR`.

**Comment by nox** [comment]
Can we unstuck this to progress on the servo bump?

**Comment by danlrobertson** [comment]
If you need it ASAP (today) I can pull the tests out and submit a separate PR for that... Otherwise I've got to do some work on that to make it work on Mac. I'd rather keep it as one PR but I totally understand if you need it ASAP

**Comment by nox** [comment]
Beginning of next week is fine too.
